app/templates/homepage.html

```diff
@@ -1,4 +1,17 @@
 <main>
+    <div class="modal fade" id="usernameModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-body">
+                <form ng-submit="homepage.setUsername(newUsername)">
```

Comment:
Here, instead of `ng-submit="homepage.setUsername(newUsername)"` we could write `ng-submit="homepage.cookies.setUsername(newUsername)"`. This way we don't have to write an extra function in the controller and we can just call the function directly.